device/camera_reader/src/camera_reader.cpp

```diff
@@ -0,0 +1,26 @@
+#include "camera_reader/camera_reader.hpp"
+
+CameraReader::CameraReader() : Node("camera") {
+    publisher_ = image_transport::create_publisher(this, "camera/image");
+
+    // Open the default camera
+    cap_.open(0); // 0 is typically the default camera index
+
+    if (!cap_.isOpened()) {
+        RCLCPP_ERROR(this->get_logger(), "Failed to open camera");
+        return;
+    }
+
+    timer_ = this->create_wall_timer(
+        std::chrono::milliseconds(100), std::bind(&CameraReader::timer_callback, this)
+    );
+}
+
+void CameraReader::timer_callback() {
+    cv::Mat frame = cap_.read();
```

Comment:
https://docs.opencv.org/4.x/d8/dfe/classcv_1_1VideoCapture.html#a473055e77dd7faa4d26d686226b292c1

```suggestion
    cv::Mat frame;
    if (!cap_.read(frame)) {
        return;
    }
```